if no vendor has no interfaces, show informative message where table rows would be (close 637 if this is merged)

jmartin-sul opened the pull request:
i didn't see unit tests for this page, so i just tested manually and took screenshots, but happy to revisit that if desired.

closes #590

2/2 slightly different implementations.  this one deviates a bit more from what the ticket literally asked for, but i think is more consistent with the table headers and empty table message we show in a similar situation, when there are no vendors to list (e.g. /vendor_management/vendors?filter=active_interfaces when no interfaces are active).

fwiw, the ticket didn't ask for a placeholder message at all, but it seemed like some text would be helpful?

before:
<img width="1135" alt="Screen Shot 2023-06-13 at 12 24 55 AM" src="https://github.com/sul-dlss/libsys-airflow/assets/7741604/e0ce482c-a13d-4d96-ad7f-ba16a7ac3e48">


after:
<img width="1131" alt="Screen Shot 2023-06-13 at 12 26 12 AM" src="https://github.com/sul-dlss/libsys-airflow/assets/7741604/90a0bd29-a179-428c-9609-3ed928b0bf34">


still works when there are interfaces:
<img width="1132" alt="Screen Shot 2023-06-13 at 12 29 41 AM" src="https://github.com/sul-dlss/libsys-airflow/assets/7741604/88058950-93ad-4604-b28c-8977f12517c4">

justinlittman commented on libsys_airflow/plugins/vendor_app/templates/vendors/vendor.html:
```diff
@@ -85,29 +85,33 @@ <h2>Vendor Interfaces</h2>
-        </td>
-    </tr>
-    {% endfor %}
+    {% if vendor.vendor_interfaces | length > 0 %}
```
I don't think you need `> 0`.

justinlittman reviewed (approved):
Approve with a non-blocking comment for your consideration.

mjgiarlo commented on libsys_airflow/plugins/vendor_app/templates/vendors/vendor.html:
```diff
@@ -85,29 +85,33 @@ <h2>Vendor Interfaces</h2>
+            </tr>
+        {% endfor %}
+    {% else %}
+        <tr><td colspan="7" class="text-center"><em>No interfaces configured for {{ vendor.display_name }}</em></td class="datatable-empty" colspan="7"></tr>
```
`</td class="datatable-empty" colspan="7">` looks like it may have some copypasta in it.

jmartin-sul replied:
ha, yes, good catch!  will remove.

jmartin-sul commented:
ok, pushed up this change, which hopefully addresses comments from @justinlittman and @mjgiarlo, will merge once green since it was approved pending fixups:

```diff
% git diff origin/iss590-dont-show-vendor-interfaces-table-when-vendor-has-no-interfaces-alt HEAD 
diff --git a/libsys_airflow/plugins/vendor_app/templates/vendors/vendor.html b/libsys_airflow/plugins/vendor_app/templates/vendors/vendor.html
index 0e0a88b..3f581a6 100644
--- a/libsys_airflow/plugins/vendor_app/templates/vendors/vendor.html
+++ b/libsys_airflow/plugins/vendor_app/templates/vendors/vendor.html
@@ -85,7 +85,7 @@
         <th>Processing DAG</th>
         <th>Inactive/Active</th>
     </thead>
-    {% if vendor.vendor_interfaces | length > 0 %}
+    {% if vendor.vendor_interfaces %}
         {% for interface in vendor.vendor_interfaces %}
             <tr>
                 <td>
@@ -110,7 +110,7 @@
             </tr>
         {% endfor %}
     {% else %}
-        <tr><td colspan="7" class="text-center"><em>No interfaces configured for {{ vendor.display_name }}</em></td class="datatable-empty" colspan="7"></tr>
+        <tr><td colspan="7" class="text-center"><em>No interfaces configured for {{ vendor.display_name }}</em></td></tr>
     {% endif %}
 </table>
```

re-tested locally, and behavior was same as in the screenshots.
